### ISSUE TITLE
Mutating webhook approach only works for pods exposing a single port

### ISSUE THREAD

**Issue by maxlegault** [telepresenceio/telepresence#1846]
The mutating webhook approach that injects the Traffic Agent based on the `telepresence.getambassador.io/inject-traffic-agent: enabled` annotation is currently limited to pods that expose a single port. As soon as there are multiple ones, the webhook will fail to inject the Traffic Agent because it can't resolve which port should be intercepted. Since it's a common use case for a pod to expose multiple ports (for example one for the main API and one for metrics), we should have a mechanism that allow using the mutating webhook in such a state.

Having talked with @thallgren about this, one quick solution could be to add an annotation to specify which port to intercept along with the `inject-traffic-agent` one. Although this would unblock using the webhook, it would limit the interceptable port of that workload to the one specified in the annotation. This means that in a GitOps flow that the webhook is helping to solve, I would need to first open a PR to specify which port I want to intercept and then have it merged so that the GitOps desired state is reflected in my cluster before I can do an intercept when I decide to intercept a different port than the one previously configured.

An ideal solution for me, but that would probably require a bigger redesign of the architecture, would be for the Traffic Agent to intercept all exposed ports in the pod and route them to the container or a workstation depending on the current live intercepts. This would also enable multiple users to simulatneously intercept different ports on the same service.

**Comment by maxlegault** [comment]
This is a duplicate of https://github.com/telepresenceio/telepresence/issues/1796. Closing this as it's been solved by the `telepresence.getambassador.io/inject-service-port` annotation. The Traffic Agent managing multiple ports should be handled in another issue.
